File: app/src/main/java/com/github/malitsplus/shizurunotes/ui/clanbattle/ClanBattleViewModel.java
package com.github.malitsplus.shizurunotes.ui.clanbattle;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ClanBattleViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public ClanBattleViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is gallery fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/github/malitsplus/shizurunotes/ui/clanbattle/ClanBattleViewModel.kt
package com.github.malitsplus.shizurunotes.ui.clanbattle

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.github.malitsplus.shizurunotes.data.ClanBattlePeriod

class ClanBattleViewModel : ViewModel() {

    val periodList = MutableLiveData<List<ClanBattlePeriod>>()


}